lib/SIL/Verifier/LinearLifetimeChecker.cpp

```diff
@@ -307,7 +307,18 @@ void State::checkForSameBlockUseAfterFree(Operand *consumingUse,
   for (auto *nonConsumingUse : nonConsumingUsesInBlock) {
     if (std::find_if(consumingUse->getUser()->getIterator(), userBlock->end(),
                      [&nonConsumingUse](const SILInstruction &i) -> bool {
-                       return nonConsumingUse->getUser() == &i;
+                       if (nonConsumingUse->getUser() != &i) {
+                         return false;
+                       }
+                       if (!isa<TermInst>(&i)) {
+                         return true;
+                       }
+                       // guaranteed phi args are verifier as part of
+                       // ReborrowVerifier
+                       if (nonConsumingUse->get().getOwnershipKind() ==
```

Comment:
I don't understand why this is here. This is a detail of the ReborrowVerifier leaking into the LinearLifetimeChecker. Can you elaborate here?